From 481606fff85b1dbf9ddd31bb8e67a747c613af81 Mon Sep 17 00:00:00 2001
Message-Id: <481606fff85b1dbf9ddd31bb8e67a747c613af81.1417780878.git.chang-joon.lee@intel.com>
In-Reply-To: <cb098d33cdac3271103834fbce08218248d7bddb.1417780878.git.chang-joon.lee@intel.com>
References: <cb098d33cdac3271103834fbce08218248d7bddb.1417780878.git.chang-joon.lee@intel.com>
From: Vandita Kulkarni <vandita.kulkarni@intel.com>
Date: Sat, 15 Nov 2014 13:44:49 +0530
Subject: [PATCH 23/34] SQUASHME! [VPG]: drm/i915: For non-atomic paths check
 for maxfifo update in primary plane enable

SQUASHME! - This patch should be squashed into the following existing patch:
	Author: Arun R Murthy <arun.r.murthy@intel.com>
	Change-Id: Id9fb3280aa59680a27536658dcaa335504a02e31
	REVERTME [VPG]: drm/i915: Add atomicity to i915.

This patch is needed for non-atomic path. maxfifo handling is required in
enable primary plane as well to avoid flicker.

Issue: GMIN-3045
Change-Id: I5e5df45fdf32d77bb9326033d78a43de5506c4c1
Signed-off-by: vandita kulkarni <vandita.kulkarni@intel.com>
Signed-off-by: Vidya Srinivas <vidya.srinivas@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |    9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 23429f1..3a71138 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -2206,6 +2206,15 @@ static void intel_enable_primary_hw_plane(struct drm_i915_private *dev_priv,
 
 	dev_priv->plane_stat |= VLV_UPDATEPLANE_STAT_PRIM_PER_PIPE(pipe);
 
+	/*
+	 * Since we are enabling a plane, we
+	 * need to make sure that we do not keep the
+	 * maxfifo enabled, if we already have one plane
+	 * enabled
+	 */
+	if (!dev_priv->atomic_update)
+		intel_update_maxfifo(dev_priv);
+
 	reg = DSPCNTR(plane);
 	val = I915_READ(reg);
 	WARN_ON(val & DISPLAY_PLANE_ENABLE);
-- 
1.7.9.5

